src/background/services/monetization.ts

```diff
@@ -170,19 +174,19 @@ export class MonetizationService {
   }
 
   async resumePaymentSessionsByTabId(tabId: number) {
-    const { enabled, connected } = await this.storage.get([
-      'connected',
-      'enabled'
-    ])
-    if (connected === false || enabled === false) return
-
     const sessions = this.sessions[tabId]
-
-    if (!sessions) {
+    if (!sessions?.size) {
       this.logger.debug(`No active sessions found for tab ${tabId}.`)
       return
     }
 
+    const { state, connected, enabled } = await this.storage.get([
+      'state',
+      'connected',
+      'enabled'
+    ])
+    if (state !== null || !connected || !enabled) return
+
```

Comment:
TODO: refactor this into `canMakePayment` something. Also, for better performance, store state, connected and enabled as member fields, and not access from storage each time (making sure they're synced of course).

Reply:
Add a todo comment or log an issue for this, please. 🙏